### PR TITLE
feat(npm): improvements

### PR DESCRIPTION
**Pull request by harimm**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Improve completion spec of `npm`

**What is the current behavior? (You can also link to an open issue here)**
Several commands are incorrectly grouped. Several options are missing
Partially fixes issue: https://github.com/withfig/autocomplete/issues/983

**What is the new behavior (if this is a feature change)?**
Commands grouped together and options added

**Additional info:**
NOTE: Though this improves the whole npm spec by adding options to certain subcommands and by grouping aliased subcommands together, this is in no way exhaustive and further improvements would be needed

### CODE REVIEW

File: src/npm.ts [hunk 1]
```diff
@@ -462,6 +470,18 @@ const completionSpec: Fig.Spec = {
...
+        description:
+          "If the fix argument is provided, then remediations will be applied to the package tree",
+        suggestions: [
+          {
+            name: "fix",
+            description:
+              "If the fix argument is provided, then remediations will be applied to the package tree",
+          },
+        ],
+      },
```
Comment (fedeci) [inline src/npm.ts:484]: This needs to be added as a subcommand

File: src/npm.ts [hunk 2]
```diff
@@ -485,42 +505,52 @@ const completionSpec: Fig.Spec = {
...
+      name: ["bugs", "issues"],
+      description: "Report bugs for a package in a web browser",
+      args: {
+        name: "pkgname",
+        isOptional: true,
+      },
       options: [
         {
-          name: "-g",
+          name: "--no-browser",
```
Comment (fedeci) [inline src/npm.ts:526]: This is `exclusiveOn` `browser`

File: src/npm.ts [hunk 3]
```diff
@@ -485,42 +505,52 @@ const completionSpec: Fig.Spec = {
...
+          description: "Display in command line instead of browser",
+        },
+        {
+          name: "--browser",
```
Comment (fedeci) [inline src/npm.ts:530]: This is `exclusiveOn` `--no-browser`

File: src/npm.ts [hunk 4]
```diff
@@ -531,22 +561,43 @@ const completionSpec: Fig.Spec = {
...
+          args: {
+            name: "audit",
+            suggestions: ["true", "false"],
+          },
+        },
+        {
+          name: "--no-audit",
+          description:
+            "Do not submit audit reports alongside the current npm command",
+        },
```
Comment (fedeci) [inline src/npm.ts:581]: These are also mutually exclusive

**Review by fedeci — CHANGES_REQUESTED** [review]
A couple things to adjust but this looks really good, thanks for cleaning up this spec a bit!

File: src/npm.ts (hunk 1 above)
Comment (harimm) [reply src/npm.ts]: Done

File: src/npm.ts (hunk 2 above)
Comment (harimm) [reply src/npm.ts:535]: The `exclusiveOn` functionality is really impressive. Fixed!

File: src/npm.ts (hunk 3 above)
Comment (harimm) [reply src/npm.ts:540]: Fixed

File: src/npm.ts (hunk 4 above)
Comment (harimm) [reply src/npm.ts]: Fixed

**Comment by harimm** [comment]
Thanks @fedeci . I have addressed review comments

**Review by harimm — COMMENTED** [review]
All suggested changes have been made. Thanks!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**